tests/core/unittests/mongodb_test.py

```diff
@@ -182,15 +188,36 @@ def test_update_many_default(self, database, moptdb):
         assert value[1]['pool_size'] == 16
         assert value[2]['pool_size'] == 2
 
+    def test_update_with_id(self, exp_config, database, moptdb):
```

Comment:
You should also test when using a fake ```_id``` that is not present in db. MongoDB should create a new row with the given id. That's different than a bare insert.